pydeseq2/dds.py

```diff
@@ -226,55 +235,57 @@ def __init__(
             )
 
         self.fit_type = fit_type
+        self.design = design
+        self.factor_storage = None
+        self.variable_to_factors = None
 
-        # Convert design_factors to list if a single string was provided.
-        self.design_factors = (
-            [design_factors] if isinstance(design_factors, str) else design_factors
-        )
-        self.continuous_factors = continuous_factors
-
-        if self.obs[self.design_factors].isna().any().any():
-            raise ValueError("NaNs are not allowed in the design factors.")
-        self.obs[self.design_factors] = self.obs[self.design_factors].astype(str)
-
-        # Check that design factors don't contain underscores. If so, convert them to
-        # hyphens.
-        if np.any(["_" in factor for factor in self.design_factors]):
+        if continuous_factors is not None:
             warnings.warn(
-                """Same factor names in the design contain underscores ('_'). They will
-                be converted to hyphens ('-').""",
-                UserWarning,
+                "continuous_factors is deprecated and will soon be removed."
+                "Continuous factors are now automatically detected from the design,"
+                "or can be cast to categorical using the C() operator in the formula",
+                DeprecationWarning,
```

Comment:
For backwards compatibility, consider adding something like

```python
design_factors.extend(
    [f"C({f})" for f in continuous_factors]
)
```

Reply:
Not sure how to ensure backwards compatibility, because pydeseq2 currently considers all factors as categorical unless passed to `continuous_factors`, whereas `formulaic` automatically detects continuous (numerical) factors but implements the `C()` operator to force categorical encoding. In a way, these are opposite behaviours.

The closest I can think of would be something like 
```
self.design_factors = [f"{f}" if f in continuous_factors else f"C({f})" for f in design_factors]
```
but that doesn't ensure that factors listed in `continuous_factors` are indeed treated as numerical, only that those which aren't are treated as categorical.

Reply:
my bad, I think you can just add them to the design factors.